api/models.py

```diff
@@ -247,7 +265,37 @@ def translate_fields(cls, params: dict):
         parent = params.get('parent')
         if parent:
             translated['parent'] = ObjectId(parent)
-        return translated
+
+        timestamp_fields = ('created', 'updated', 'timeout', 'holdoff')
+
+        operator_map = {
+            'lt': '$lt',
+            'lte': '$lte',
+            'gt': '$gt',
+            'gte': '$gte',
+        }
+
+        for key in params.keys():
+            if key in timestamp_fields:
+                translated[key] = datetime.fromisoformat(translated[key])
+            if '__' in key:
+                field, operator = key.split('__')
```

Comment:
If `__` is twice in the key then it won't really work.  I think it's better to just call `split` without an `if` and then check the returned value.

Reply:
This should be a simple rework, do you see any issue with it?

Reply:
Yes, I'll fix this.